### ISSUE TITLE
Tests Needed for Command Line Arg Processing with CALL

### ISSUE THREAD

**Issue by hostilefork** [metaeducation/ren-c#996]
Red has looked into command-line escaping cases and put together a table of behaviors:

https://github.com/red/red/pull/3870

These should be double-checked, and if correct, the behaviors verified and added to the tests (using CALL of a command-line echoer script).

(@draegtun - you've looked into this sort of thing before with CALL and the tests, perhaps it could be an interesting thing to look at...?  Just bringing it to your attention...)

Here is also some long-ago discussion on command line argument processing to weigh in on building some tests:

https://github.com/metaeducation/ren-c/issues/46

**Comment by draegtun** [comment]
re: CLI tests - Yes i have a branch from July 2017 which I've never committed (even locally until last week!).

They're incomplete and were only written to cover current behaviour so that i/we could refactor CLI parsing code.

Here's my branch - https://github.com/draegtun/ren-c/tree/test-cli/tests/command-line

If you think these are a good start then I'll look into bringing them up-to-date and doing a PR.

**Comment by hostilefork** [comment]
Sounds great!  Yeah, people complain when command line behavior doesn't meet expectations:

https://github.com/rebol/rebol/issues/246

I feel like having the commandline processing done in PARSE is the first hurdle, the C code was problematic:

https://github.com/rebol/rebol/blob/master/src/os/host-args.c

*(That may have looked simple at first glance to some people, but the devil was in the details...e.g. what's JOIN_STR or MAKE_STR, and the more you learn about that the more intractable it was...hence why it was all refactored.  But still, it's a primordial version of what it could be!)*

As a sidenote: I'm also looking at using Red's tests for "Redbol":

https://github.com/metaeducation/ren-c/blob/master/tests/red/run-red-test.reb

They name their tests, so what I imagine is that there be a directory tree which names tests that aren't expected to succeed.  This will be more.  Atronix has indicated an interest in funding Redbol development to ease their transition (don't know if you'd be looking for work, by chance...?)

**Comment by draegtun** [comment]
> Sounds great! Yeah, people complain when command line behavior doesn't meet expectations:
> 
> [rebol#246](https://github.com/rebol/rebol/issues/246)

I also agree with idanarye suggestion.  This brings it more inline with other Unix(ey) scripting languages.   It's always been annoying when piping scripts together and having a *faulty* Rebol script hang things :(

Your *"Hit Ctrl-C to break into the console in 5 seconds"* is an improvement but I would prefer if it just crashed when untrapped run-time error bubbles up in script code.  And same with `--do`

With above we could then add new command-line option to achieve old behaviour.  For eg.    `--halt-on-error`

> 
> I feel like having the commandline processing done in PARSE is the first hurdle, the C code was problematic:

Agree.  

My plan (when we last chatted about it) was to first create a general CLI arg handling module for Rebol scripts.  Next on my list after CLI tests :)

> As a sidenote: I'm also looking at using Red's tests for "Redbol":
> 
> https://github.com/metaeducation/ren-c/blob/master/tests/red/run-red-test.reb
> 
> They name their tests, so what I imagine is that there be a directory tree which names tests that aren't expected to succeed. This will be more. Atronix has indicated an interest in funding Redbol development to ease their transition (don't know if you'd be looking for work, by chance...?)

Always nice if there's an opportunity to be paid for a Rebol gig :)  

Chat more when next in chat.   In meantime I'll start looking at updating those CLI tests.
